### ISSUE TITLE
SwiftLint Compile Times Extremely Small

### ISSUE THREAD

**Issue by DomenicBianchi01** [realm/SwiftLint#5099]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

- When compiling the app with the SwiftLint SPM plugin installed, the compile times are **extremely** slow.
- It looks like the culprit is `SwiftSyntax`

![Screenshot 2023-07-05 at 10 04 15 AM](https://github.com/realm/SwiftLint/assets/25673485/27dcaf43-07ec-4733-83a9-b9aa6fe7b7d8)

### Environment

* SwiftLint version: 0.52.3
* Xcode version: 14.3.1
* Installation method: SPM

**Comment by DomenicBianchi01** [comment]
Related issue: https://github.com/realm/SwiftLint/issues/4261

**Comment by marcelofabri** [comment]
Not really related, that issue is about the runtime performance of SwiftLint. Optimizing for compilation times hasn't been a goal of SwiftLint. IMO the real issue here is how SPM works if this is happening in every build.

I personally always recommend people using a pre-built binary if they don't want to pay the compilation cost.

**Comment by jpsim** [comment]
You're correct to note that SwiftSyntax is responsible for the bulk of the compilation time. That's also tracked here: https://github.com/apple/swift-syntax/issues/1194

This is why most users should be using pre-built SwiftLint binaries, which the SwiftPM plugin should be using by default. If that's not happening for you for some reason, please share more information.

**Comment by anton-plebanovich** [comment]
Not sure if related or if should I create a separate issue. I'm facing an issue of huge lint times when I try to pass concrete files to lint. In my project, it takes ~1 second to perform full lint of ~250 files and ~12 seconds when I pass ~60 files. Version 0.52.4. 

It takes ~1 second on Version 0.51.0 for example for the same 60 files.

My teammate on a different project reported ~10 mins lint time on bigger file counts and on slower machines.

**Comment by jpsim** [comment]
How does that compare with running `swiftlint` as installed from Homebrew from your command line?

**Comment by anton-plebanovich** [comment]
> How does that compare with running swiftlint as installed from Homebrew from your command line?
If that's a question for me, I use Cocoapods and binary there so not sure if it makes sense to check binary from Homebrew.

**Comment by Mordil** [comment]
We made an upgrade to one of our projects that necessitated embedding & committing almost 100 new files to our project that previously were codegen as part of a SwiftPM build plugin.

Even after adding a globstar pattern to ignore those files, we're seeing SwiftLint go from 3s tops to now 150s in some cases.

This is with the Swift Plugin version of SwiftLint, on both 0.51 and 0.52

We have 24 opt-in rules enabled, and 12 rules turned off. 1 rule is deprecated - `unused_capture_list`

We do have the `explicit_self` analyzer rule enabled as well, but even disabling it we're still seeing the massive increase in runtime

The following screenshots are with one with the analyzer rule, and without

| analyzer ON | analyzer OFF |
|-------------|--------------|
| ![Screenshot 2023-08-03 at 12 39 10](https://github.com/realm/SwiftLint/assets/3506776/f55ec9c3-b124-430b-92c6-a000bca2771a) | ![Screenshot 2023-08-03 at 12 39 02](https://github.com/realm/SwiftLint/assets/3506776/23b54a22-47b5-460a-b323-36bb97656a60) |

**Comment by Mordil** [comment]
Ah, it looks like my comment likely relates to https://github.com/realm/SwiftLint/issues/5018 instead

**Comment by fl034** [comment]
> You're correct to note that SwiftSyntax is responsible for the bulk of the compilation time. That's also tracked here: [apple/swift-syntax#1194](https://github.com/apple/swift-syntax/issues/1194)
> 
> This is why most users should be using pre-built SwiftLint binaries, which the SwiftPM plugin should be using by default. If that's not happening for you for some reason, please share more information.

@jpsim Is there a way to offer a Swiftlint binary as Xcode plugin? 
In a Swift package, using the Xcode plugin seems to be the only way to lint the code.
